server/file_handler.cc

```diff
@@ -0,0 +1,59 @@
+#include "file_handler.h"
+#include <vector>
+#include <string.h>
+#include <cstdio>
+#include "http.h"
+
+RequestHandler::Status StaticFileHandler::Init(const std::string& uri_prefix, const NginxConfig& config){
+	this->m_uri_prefix_ = uri_prefix;
+	//assume the config argument has something like
+	// root /path1
+	//check if such a line exists
+	auto root_config_tokens = config.find("root");
```

Comment:
I'm not a big fan of using auto here. Automatically inferring types isn't a good practice